jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/Invocable.java

```diff
@@ -61,12 +61,17 @@ enum InvocationType
          */
         NON_BLOCKING,
         /**
-         * <p>Invoking the {@link Invocable} may block the invoker thread,
-         * but the invocation cannot be deferred to a later time, differently
-         * from {@link #BLOCKING}.</p>
+         * <p>Invoking the {@link Invocable} may either block the invoker thread if
+         * called directly, or must not block the invoker thread if called via
+         * {@link Invocable#invokeNonBlocking(Runnable)}</p>
          * <p>This invocation type is suitable for {@code Invocable}s that
-         * themselves perform the non-deferrable action in a non-blocking way,
-         * thus advancing a possibly stalled system.</p>
+         * may contain both high non-blocking and low priority blocking sub tasks.
+         * Because they may contain high priority tasks, the invocation may not be deferred.
+         * If calling threads are limited, then the task can be invoked as non-blocking so
+         * that the high priority tasks will be invoked. The low priority tasks may be
+         * scheduled to be executed later.   However, if threads are not constrained, then
+         * the task can be called directly, which may directly call a low priority blocking
+         * task as an optimization.</p>
```

Comment:
I feel like the subtasks example is not capturing the generic meaning, and the reference to the number of threads further complicates things.

I'll try to come up with something alternative.

Reply:
@sbordet Hmmm not so much.  That is describing how we are specifically using the mechanism, but it is not describing the mechanism.  If we are doing this so that 3rd parties can create EITHER tasks, then they may not use a produce/consume style implementation.
I guess that applies for the discussion of threads in my version to.... let me have a think.

Reply:
I've rewritten without as much reference to constrained thread numbers